src/graph/ConfigExecutor.cpp

```diff
@@ -0,0 +1,318 @@
+            space = *configItem_->getSpace();
+        }
+        if (configItem_->getModule() != nullptr) {
+            module = *configItem_->getModule();
+        }
+        if (configItem_->getName() != nullptr) {
+            name = *configItem_->getName();
+        }
+        if (configItem_->getValue() != nullptr) {
+            value = configItem_->getValue()->eval();
+        }
+    }
+
+    auto future = ectx()->configManager()->setConfig(space, module, name, value);
+    auto *runner = ectx()->rctx()->runner();
+
+    auto cb = [this] (auto && resp) {
+        if (!resp.ok()) {
+            DCHECK(onError_);
+            onError_(std::move(resp).status());
+            return;
+        }
+
+        resp_ = std::make_unique<cpp2::ExecutionResponse>();
+
+        DCHECK(onFinish_);
+        onFinish_();
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        DCHECK(onError_);
+        onError_(Status::Error(folly::stringPrintf("Internal error : %s",
+                                                   e.what().c_str())));
+        return;
+    };
+    std::move(future).via(runner).thenValue(cb).thenError(error);
+}
+
+void ConfigExecutor::getVariables() {
+    std::string space;
+    std::string module;
+    std::string name;
+    ColumnType type;
+    if (configItem_ != nullptr) {
+        if (configItem_->getSpace() != nullptr) {
+            space = *configItem_->getSpace();
+        }
+        if (configItem_->getModule() != nullptr) {
+            module = *configItem_->getModule();
+        }
+        if (configItem_->getName() != nullptr) {
+            name = *configItem_->getName();
+        }
+        if (configItem_->getType() != nullptr) {
+            type = *configItem_->getType();
+        }
+    }
+
+    auto future = [&] () -> folly::Future<StatusOr<meta::ConfigItem>> {
+        switch (type) {
+            case ColumnType::INT:
+                return ectx()->configManager()->getConfigAsInt64(space, module, name);
+            case ColumnType::DOUBLE:
+                return ectx()->configManager()->getConfigAsDouble(space, module, name);
+            case ColumnType::BOOL:
+                return ectx()->configManager()->getConfigAsBool(space, module, name);
+            case ColumnType::STRING:
+                return ectx()->configManager()->getConfigAsString(space, module, name);
+            default:
+                return Status::Error("parse value type error");
+        }
+    }();
+    auto *runner = ectx()->rctx()->runner();
+
+    auto cb = [this] (auto && resp) {
+        if (!resp.ok()) {
+            DCHECK(onError_);
+            onError_(std::move(resp).status());
+            return;
+        }
+
+        std::vector<std::string> header;
+        header.push_back("space");
+        header.push_back("module");
+        header.push_back("name");
+        header.push_back("type");
+        header.push_back("value");
```

Comment:
use initialize list maybe more performance like std::vector<std::string> header({"a", "b", "c"})

Reply:
Yeah, you're right, roger that!